src/unit_tests/test_json_queue.c

```diff
@@ -0,0 +1,1070 @@
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "../headers/shared.h"
+
+static int test_mode = 0;
+
+int Handle_JQueue(file_queue *fileq, int flags) __attribute__((nonnull));
+
+int __wrap__minfo()
+{
+    return 0;
+}
+
+void __wrap__merror(const char * file, int line, const char * func, const char *msg, ...)
+{
+    char formatted_msg[OS_MAXSTR];
+    va_list args;
+
+    va_start(args, msg);
+    vsnprintf(formatted_msg, OS_MAXSTR, msg, args);
+    va_end(args);
+
+    check_expected(formatted_msg);
+}
+
+extern FILE * __real_fopen(const char * __filename, const char * __modes);
+FILE * __wrap_fopen(const char * __filename, const char * __modes)
+{
+    if (test_mode) {
+        check_expected(__filename);
+        check_expected(__modes);
+
+        return mock_type(FILE *);
+    } else {
+        return __real_fopen(__filename, __modes);
+    }
+}
+
+extern int __real_fclose (FILE *__stream);
+int __wrap_fclose(FILE *__stream)
+{
+    if (test_mode) {
+        check_expected_ptr(__stream);
+
+        return 0;
+    } else {
+        return __real_fclose(__stream);
+    }
+}
+
+extern int __real_fseek(FILE *__stream, long int __off, int __whence);
+int __wrap_fseek(FILE *__stream, long int __off, int __whence)
+{
+    if (test_mode) {
+        return mock();
+    } else {
+        return __real_fseek(__stream, __off, __whence);
+    }
+}
+
+extern int __real_fstat (int __fd, struct stat *__buf);
+int __wrap_fstat(int __fd, struct stat *__buf)
+{
+    if (test_mode) {
+        return mock();
+    } else {
+        return __real_fstat(__fd, __buf);
+    }
+}
+
+void __wrap_fileno()
+{
+    return;
+}
+
+void __wrap_clearerr()
+{
+    return;
+}
+
+char * __wrap_fgets(char * __s, int __n, FILE * __stream)
+{
+    check_expected(__n);
+    check_expected_ptr(__stream);
+
+    strcpy(__s, mock_type(char *));
+
+    return mock_type(char *);
+}
+
+static int init_test_mode(void **state)
+{
+    (void) state;
+
+    test_mode = 1;
+
+    return 0;
+}
+
+static int end_test_mode(void **state)
+{
+    (void) state;
+
+    test_mode = 0;
+
+    return 0;
+}
+
+void test_jqueue_init(void **state)
+{
+    (void) state;
+
+    file_queue *fileq;
+
+    os_calloc(1, sizeof(file_queue), fileq);
+
+    jqueue_init(fileq);
+
+    assert_int_equal(fileq->last_change, 0);
+    assert_int_equal(fileq->year, 0);
+    assert_int_equal(fileq->day, 0);
+    assert_int_equal(fileq->flags, 0);
+    assert_null(fileq->mon[0]);
+    assert_null(fileq->file_name[0]);
+    assert_null(fileq->fp);
+
+    free(fileq);
+}
+
+void test_jqueue_open_fail_fopen(void **state)
+{
+    (void) state;
+
+    file_queue *fileq;
+
+    os_calloc(1, sizeof(file_queue), fileq);
+
+    /* fopen fail */
+
+    jqueue_init(fileq);
```

Comment:
I would move the os_calloc and jqueue_init into a setup function for this test and pass it via `state`.
The ending free statement should also be moved into a teardown function, by doing this we guarantee that the alloc'd memory will be freed regardless of the test passing or failing.
Also, replace calls to wazuh specific code (such as os_calloc) with std function calls.

Reply:
Done!